### PR TITLE
Create daemon handler fixture for integration test

### LINKED ISSUE #1825: Wazuh daemons handler fixture
### Description

Currently, there are multiple fixtures in different paths to restart every Wazuh daemon. This is a duplicated code, and it should be desirable to refactorize.
Also, due to the change of the [development guidelines](https://github.com/wazuh/wazuh-qa/wiki/Development-guidelines) of the repository,  Wazuh new default status is off, making necessary the creation of new tools to apply this change.

For this reason, is necessary to create a global fixture to handle wazuh daemons according to the repository standards

### PR DESCRIPTION
**Pull request by Rebits**
|Related issue|
|---|
|#1825|

## Description

This PR closes #1825. It adds a common Wazuh daemon handler fixture for all integration tests, according to the [repository standard](https://github.com/wazuh/wazuh-qa/wiki/Development-guidelines)

## Usage

In order to use this fixture, the test should define the following dictionary at the beginning of the module:

```
daemons_handler_configuration = {'daemons': ['wazuh-remoted'], 'ignore_errors': True}
```

In case of the required of restart all wazuh, it is possible to use wazuh_control using the following configuration

```
daemons_handler_configuration = {'wazuh_control': True, 'ignore_errors': True}
```

Where:
- `daemons` (list): List of daemons to use. If empty, all wazuh will be used.
- `wazuh_control` (boolean): Enable restart of all wazuh services using wazuh_control.
- `ignore_errors` (boolean): Enable ignore errors in daemon handling process.

## Tests

- [x] Proven that this fixture works correctly

- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.

### CODE REVIEW

**Comment by Selutario** [comment]
### Additional info

In the case of creating a variable with the necessary daemons for the API to work, the list is as follows: 
- `wazuh-apid`
- `wazuh-modulesd`
- `wazuh-analysisd`
- `wazuh-execd`
- `wazuh-db`
- `wazuh-remoted`

If any of those daemons are not ready, the error below will be returned for any API request:
```JSON
{
  "title": "Bad Request",
  "detail": "Some Wazuh daemons are not ready yet in node \"master-node\" (wazuh-modulesd->stopped, wazuh-analysisd->stopped, wazuh-execd->stopped, wazuh-remoted->stopped)",
  "dapi_errors": {
    "master-node": {
      "error": "Some Wazuh daemons are not ready yet in node \"master-node\" (wazuh-modulesd->stopped, wazuh-analysisd->stopped, wazuh-execd->stopped, wazuh-remoted->stopped)"
    }
  },
  "error": 1017
}
```

Regards.

File: tests/integration/conftest.py [hunk 1]
```diff
@@ -602,3 +602,74 @@ def create_file_structure_function(get_files_list):
...
+@pytest.fixture(scope='module')
+def daemons_handler(get_configuration, request):
+    """Handler of Wazuh daemons.
+
+    It uses `daemons_handler_configuration` of each module in order to configure the behavior of the fixture.
+    The  `daemons_handler_configuration` should be a dictionary with the following keys:
+        daemons (list, optional): List with every daemon to be used by the module. In case of empty a ValueError
+            will be raised
+        wazuh_control (boolean): Configure to restart all wazuh services. Default `False`.
+        ignore_errors (boolean): Configure if errors in daemon handling should be ignore. This option is available
```
Comment (jmv74211) [inline tests/integration/conftest.py:616]:
```suggestion
        ignore_errors (boolean): Configure if errors in daemon handling should be ignored. This option is available
```

File: tests/integration/conftest.py [hunk 2]
```diff
@@ -602,3 +602,74 @@ def create_file_structure_function(get_files_list):
...
+    """
+    daemons = []
+    ignore_errors = False
+    wazuh_control = False
+
+    try:
+        daemons_handler_configuration = getattr(request.module, 'daemons_handler_configuration')
+        if 'daemons' in daemons_handler_configuration and not wazuh_control:
+            daemons = daemons_handler_configuration['daemons']
+            if not daemons:
```
Comment (jmv74211) [inline tests/integration/conftest.py:631]: Test this

```suggestion
            if not daemons or (type(daemons) == list and len(daemons) == 0):
```

File: tests/integration/conftest.py [hunk 3]
```diff
@@ -602,3 +602,74 @@ def create_file_structure_function(get_files_list):
...
+    try:
+        if wazuh_control:
+            logger.debug('Daemons Handler: Debug - Restarting wazuh using wazuh-control')
+            # Restart daemon instead of starting due to legacy used fixture in the test suite.
+            control_service('restart')
+        else:
+            for daemon in daemons:
+                logger.debug(f"Daemons Handler: Restarting {daemon}")
+                # Restart daemon instead of starting due to legacy used fixture in the test suite.
+                control_service('restart', daemon=daemon)
```
Comment (jmv74211) [inline tests/integration/conftest.py:656]: What happens if we have entered a daemon name that does not exist or is invalid?

File: tests/integration/conftest.py [hunk 4]
```diff
@@ -602,3 +602,74 @@ def create_file_structure_function(get_files_list):
...
+
+    yield
+
+    if wazuh_control:
+        logger.debug('Daemons Handler: Debug - Stopping wazuh using wazuh-control')
+        control_service('stop')
+    else:
+        for daemon in daemons:
+            logger.debug(f"Daemons Handler: Stopping {daemon}")
+            control_service('stop', daemon=daemon)
```
Comment (jmv74211) [inline tests/integration/conftest.py:675]: New line at the end of the file?

**Comment by Rebits** [comment]
Respecting to [this](https://github.com/wazuh/wazuh-qa/pull/1826#pullrequestreview-744921021) comment, yes, that could be an issue, but IMHO it is important to begin to include some verbose in our testing procedure. Maybe it would be necessary to create a new logs levels and implement tools to handle the `log-cli-level`, `log-file-level` and `log-file` of pytest.

File: tests/integration/conftest.py (hunk 3 above)
Comment (Rebits) [reply tests/integration/conftest.py:656]: In that case, the test will raise an error at this `contol_service` line

https://github.com/wazuh/wazuh-qa/blob/f32574624734736870596be059d88722e938ba85/deps/wazuh_testing/wazuh_testing/tools/services.py#L126

Maybe this function needs a refactor in which we can include more verbose and daemons values handling.

Comment (Rebits) [reply tests/integration/conftest.py:656]: Done  52b1d67

File: tests/integration/conftest.py (hunk 2 above)
Comment (Rebits) [reply tests/integration/conftest.py]: Done 52b1d67

File: tests/integration/conftest.py (hunk 1 above)
Comment (Rebits) [reply tests/integration/conftest.py]: Done 52b1d67

File: deps/wazuh_testing/wazuh_testing/tools/__init__.py [hunk 5]
```diff
@@ -146,3 +150,27 @@ def get_service():
...
+MONITOR_DAEMON = 'wazuh-monitord'
+DB_DAEMON = 'wazuh-db'
+AGENT_DAEMON = 'wazuh-agentd'
+
+
+ALL_MANAGER_DAEMONS = [LOGCOLLECTOR_DAEMON, AGENTLESS_DAEMON, CSYSLOG_DAEMON, REMOTE_DAEMON, ANALYSISD_DAEMON,
+ API_DAEMON, MAIL_DAEMON, SYSCHECK_DAEMON, EXEC_DAEMON, MODULES_DAEMON, CLUSTER_DAEMON, INTEGRATOR_DAEMON, MONITOR_DAEMON,
+ DB_DAEMON ]
+ALL_AGENT_DAEMONS = [AGENT_DAEMON, EXEC_DAEMON, LOGCOLLECTOR_DAEMON, SYSCHECK_DAEMON, MODULES_DAEMON]
+API_DAEMON_REQUIREMENTS = [API_DAEMON, MODULES_DAEMON, ANALYSISD_DAEMON, EXEC_DAEMON, DB_DAEMON, REMOTE_DAEMON]
```
[the remark this answers is not in the document]
Comment (Rebits) [reply deps/wazuh_testing/wazuh_testing/tools/__init__.py]: Done 52b1d67

**Review by juliamagan — DISMISSED** [review]
LGTM

**Comment by juliamagan** [comment]
We can test these changes before the merge in temporal branch: `common-fixtures-testing`

File: tests/integration/conftest.py [hunk 6]
```diff
@@ -602,3 +602,74 @@ def create_file_structure_function(get_files_list):
...
+        ignore_errors (boolean): Configure if errors in daemon handling should be ignored. This option is available
+        in order to use this fixture along with invalid configuration. Default `False`
+
+    Args:
+        get_configuration (fixture): Gets the current configuration of the test.
+        request (fixture): Provide information on the executing test function.
+    """
+    daemons = []
+    ignore_errors = False
+    wazuh_control = False
```
Comment (jmv74211) [inline tests/integration/conftest.py:625]: Can you change this for `restart_all` or something similar. `wazuh_control` is a bit confusing

**Review by jmv74211 — DISMISSED** [review]

**Review by juliamagan — DISMISSED** [review]

**Review by jmv74211 — APPROVED** [review]

**Pull request merged**